src/provider.c

```diff
@@ -312,9 +312,14 @@ static int p11prov_get_capabilities(void *provctx, const char *capability,
     return RET_OSSL_ERR;
 }
 
+struct reason_string {
+    unsigned int id;
+    const char *ptr;
+};
+
 static const OSSL_ITEM *p11prov_get_reason_strings(void *provctx)
 {
-    static const OSSL_ITEM reason_strings[] = {
+    static const struct reason_string reason_strings[] = {
```

Comment:
Why this change?

Reply:
The `OSSL_ITEM` is `struct { int, void * }` and the compiler complains about dropping the `const` modifier when assigning `const char *` to `void *`.

It was either this or typecasting all strings in the array to `char *` (which also would be a valid fix). I am fine with either, but this is less chatty.

Reply:
yeah lets do a
#define C(str) (void *)(str)
before the struct and call them all as C("string")

I rather not shadow structures defined in headers.

Reply:
Done

Reply:
perfect